Cython/Compiler/ModuleNode.py

```diff
@@ -2565,71 +2551,38 @@ def generate_module_state_begin(self, env, code):
         code.putln('#endif')
 
     def generate_module_state_end(self, env, code):
-        code.putln('} %s;' % Naming.modulestate_cname)
+        code.putln('#endif /* CYTHON_USE_MODULE_STATE */')
+        code.putln('} %s;' % Naming.modulestate_type_cname)
         code.putln('')
+        code.putln('#if !CYTHON_USE_MODULE_STATE')
+        code.putln('static %s %s;' % (Naming.modulestate_type_cname, Naming.modulestate_cname))
+        code.putln('#define __Pyx_Globals() (&%s)' % Naming.modulestate_cname)
```

Comment:
Splitting things into a getter and setter pattern will be a gigantic churn. With the current approach we can leave many places in the code untouched because things naturally format for the `%s` pattern. Requiring to call explicit getters and setters means we have to explicitely review every single place in the code. I also think seeing a `=` for assignments rather than a set function helps readability.

Putting the dereferencing into the macro is fine with me. It hides more of the mechanism from the casual reader, but given that globals are used extensively all over the place that may be worth it as it reduces the noise. So how about `__Pyx_Global(name) = ...`  /  `xxx = __Pyx_Global(name)` ?   I would vote to not include the `C` into the macro name, because in the generated code it's all C anyway and there is no need to distinguish from some python name.

Reply:
I'm also open to different term than "Global" for those thing, I was contemplating stuff like "StateVar" ro "Static" but so far "Global" felt like the most natural to me.

Reply:
>  I would vote to not include the C into the macro name, because in the generated code it's all C anyway and there is no need to distinguish from some python name.

`CGlobal` feels better for me because `global` sounds a lot like "globals()". `CGlobal` makes it clear that we are not referring to the Python context. While the C code is obviously C, it comes from Python code and uses a lot of Python jargon. Also, most C code readers will be more acquainted with the Python concepts than the C concepts.

It also sounds better than `Static` because the variables are not necessarily `static`. They are in fact specific to the module instance.

Reply:
> Splitting things into a getter and setter pattern will be a gigantic churn

Maybe. Not sure. My guess is that this churn may come at some point when we switch to per module instance C globals and require a reference to `self` for looking them up. But that's best decided when we actually see what works and what doesn't.

> Requiring to call explicit getters and setters means we have to explicitely review every single place in the code.

Unless we keep the getter for the `__str__` formatting and require explicit action only for the setter. Then the C code should fail to compile in all places where a setter was intended, and those are probably few and easy to find and fix by their context. That also has the advantage that we can put in an assertion that all globals have been set *and* read at some point and are not only read, e.g. due to a typo in the name. That becomes even more interesting when we start declaring C globals automatically on use.